File: bitget-golang-sdk-api/pkg/model/mix/plan/modifyplanpresetreq.go
package plan

/**
 * @Author: bitget-sdk-team
 * @Date: 2022-09-30 10:46
 * @DES: modify plan request
 */
type ModifyPlanPresetReq struct {
	/**
	 * Currency pair
	 */
	Symbol string `json:"symbol"`
	/**
	 * Deposit currency
	 */
	MarginCoin string `json:"marginCoin"`
	/**
	 * If the profit stop price is blank, cancel the profit stop
	 */
	PresetTakeProfitPrice string `json:"presetTakeProfitPrice"`
	/**
	 * If the stop loss price is blank, cancel the stop loss
	 */
	PresetStopLossPrice string `json:"presetStopLossPrice"`
	/**
	 * order id
	 */
	OrderId string `json:"orderId"`
	/**
	 * plan type
	 */
	PlanType string `json:"planType"`
}

File: bitget-node-sdk-api/src/lib/model/mix/plan/ModifyPlanPresetReq.ts
export class ModifyPlanPresetReq{
    /**
     * Currency pair
     */
    private _symbol!:string;
    /**
     * Deposit currency
     */
    private _marginCoin!:string;
    /**
     * If the profit stop price is blank, cancel the profit stop
     */
    private _presetTakeProfitPrice!:string;
    /**
     * If the stop loss price is blank, cancel the stop loss
     */
    private _presetStopLossPrice!:string;
    /**
     * order id
     */
    private _orderId!:string;
    /**
     * plan type
     */
    private _planType!:string;


    get symbol(): string {
        return this._symbol;
    }

    set symbol(value: string) {
        this._symbol = value;
    }

    get marginCoin(): string {
        return this._marginCoin;
    }

    set marginCoin(value: string) {
        this._marginCoin = value;
    }

    get presetTakeProfitPrice(): string {
        return this._presetTakeProfitPrice;
    }

    set presetTakeProfitPrice(value: string) {
        this._presetTakeProfitPrice = value;
    }

    get presetStopLossPrice(): string {
        return this._presetStopLossPrice;
    }

    set presetStopLossPrice(value: string) {
        this._presetStopLossPrice = value;
    }

    get orderId(): string {
        return this._orderId;
    }

    set orderId(value: string) {
        this._orderId = value;
    }

    get planType(): string {
        return this._planType;
    }

    set planType(value: string) {
        this._planType = value;
    }
}